internal/builder/builder.go

```diff
@@ -103,17 +103,26 @@ type moduleWithDiffID struct {
 type BuilderOption func(*options) error
 
 type options struct {
-	flatten bool
-	exclude []string
+	flatten  bool
+	exclude  []string
+	runImage string
+}
+
+func WithRunImage(name string) BuilderOption {
+	return func(o *options) error {
+		o.runImage = name
+		return nil
+	}
 }
 
 // FromImage constructs a builder from a builder image
-func FromImage(img imgutil.Image) (*Builder, error) {
-	return constructBuilder(img, "", true)
+func FromImage(img imgutil.Image, runImage string) (*Builder, error) {
+	return constructBuilder(img, "", true, WithRunImage(runImage))
 }
 
 // New constructs a new builder from a base image
-func New(baseImage imgutil.Image, name string, ops ...BuilderOption) (*Builder, error) {
+func New(baseImage imgutil.Image, runImage string, name string, ops ...BuilderOption) (*Builder, error) {
```

Comment:
I think you need to add the `runImage` argument, the `BuilderOption` already allow you to passthrough `WithRunImage` option.